junit/junit.go

```diff
@@ -235,6 +236,16 @@ func formatDuration(d time.Duration) string {
 }
 
 // formatOutput combines the lines from the given output into a single string.
-func formatOutput(output []string, indent int) string {
-	return strings.Join(output, "\n")
```

Comment:
To keep it simple, let's escape the string after we joined the lines together. E.g.:
```go
return escapeIllegalChars(strings.Join(output, "\n"))
```